internal/indexer/block_context.go

```diff
@@ -55,7 +55,9 @@ func (bc *blockContext) process(raw *chain.Block) (dags_count, trx_count uint64,
 	tp.Go(func() { dags_count, err = bc.processDags() })
 	tp.Go(func() { err = bc.processTransactions(raw.Transactions) })
 	tp.Wait()
-
+	if bc.block.Number%1000 == 0 {
+		tp.Go(func() { err = bc.CheckIndexedBalances() })
```

Comment:
You are adding one task to thread pool and waiting for it in the next line in the next commit. So this doesn't makes sense as it still will be blocking current thread